pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1798,7 +1798,11 @@ func (l *LibvirtDomainManager) SignalShutdownVMI(vmi *v1.VirtualMachineInstance)
 	}
 
 	if domState == libvirt.DOMAIN_RUNNING || domState == libvirt.DOMAIN_PAUSED {
-		err = dom.ShutdownFlags(libvirt.DOMAIN_SHUTDOWN_ACPI_POWER_BTN)
+		flag := libvirt.DOMAIN_SHUTDOWN_DEFAULT
```

Comment:
What does this result in? The documentation states this is hypervisor specific but that's where it ends. I have looked at the libvirt code and I don't seem to find any special handling. Does it come down to qemu and if what it does?

Reply:
Just for refrence ,  shutdown and  deletion are happening both

**for virt launcher logs during shutdown:**

"component":"virt-launcher","kind":"","level":"info","msg":"Signaled graceful shutdown for testvmi-txhzt","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"manager.go:1810","timestamp":"2024-09-20T13:03:26.595310Z","uid":"c0c6908d-1f15-4f0d-9e9c-c33a1f766439"}
{"component":"virt-launcher","kind":"VirtualMachineInstance","level":"info","msg":"Signaled graceful shutdown","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"virt-launcher.go:443","timestamp":"2024-09-20T13:03:26.599302Z","uid":"c0c6908d-1f15-4f0d-9e9c-c33a1f766439"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Signaled vmi shutdown","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"server.go:345","timestamp":"2024-09-20T13:03:26.615246Z","uid":"c0c6908d-1f15-4f0d-9e9c-c33a1f766439"}
{"component":"virt-launcher","level":"info","msg":"kubevirt domain status: Running(1):Unknown(1)","pos":"client.go:297","timestamp":"2024-09-20T13:03:26.624539Z"}
={"component":"virt-launcher","level":"info","msg":"kubevirt domain status: Running(1):Unknown(1)","pos":"client.go:297","timestamp":"2024-09-20T13:03:26.680155Z"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Signaled graceful shutdown for testvmi-txhzt","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"manager.go:1810","timestamp":"2024-09-20T13:03:26.698424Z","uid":"c0c6908d-1f15-4f0d-9e9c-c33a1f766439"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Signaled vmi shutdown","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"server.go:345","timestamp":"2024-09-20T13:03:26.699042Z","uid":"c0c6908d-1f15-4f0d-9e9c-c33a1f766439"}
={"component":"virt-launcher","kind":"","level":"info","msg":"Signaled graceful shutdown for testvmi-txhzt","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"manager.go:1810","timestamp":"2024-09-20T13:03:26.811239Z","uid":"c0c6908d-1f15-4f0d-9e9c-c33a1f766439"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Signaled vmi shutdown","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"server.go:345","timestamp":"2024-09-20T13:03:26.811441Z","uid":"c0c6908d-1f15-4f0d-9e9c-c33a1f766439"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Signaled graceful shutdown for testvmi-txhzt","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"manager.go:1810","timestamp":"2024-09-20T13:03:31.635030Z","uid":"c0c6908d-1f15-4f0d-9e9c-c33a1f766439"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Signaled vmi shutdown","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"server.go:345","timestamp":"2024-09-20T13:03:31.635302Z","uid":"c0c6908d-1f15-4f0d-9e9c-c33a1f766439"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Signaled graceful shutdown for testvmi-txhzt","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"manager.go:1810","timestamp":"2024-09-20T13:03:36.648147Z","uid":"c0c6908d-1f15-4f0d-9e9c-c33a1f766439"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Signaled vmi shutdown","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"server.go:345","timestamp":"2024-09-20T13:03:36.648411Z","uid":"c0c6908d-1f15-4f0d-9e9c-c33a1f766439"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Signaled graceful shutdown for testvmi-txhzt","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"manager.go:1810","timestamp":"2024-09-20T13:03:41.658819Z","uid":"c0c6908d-1f15-4f0d-9e9c-c33a1f766439"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Signaled vmi shutdown","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"server.go:345","timestamp":"2024-09-20T13:03:41.658975Z","uid":"c0c6908d-1f15-4f0d-9e9c-c33a1f766439"}
{"component":"virt-launcher-monitor","level":"info","msg":"Reaped pid 68 with status 0","pos":"virt-launcher-monitor.go:198","timestamp":"2024-09-20T13:03:46.806541Z"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Domain stopped.","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"manager.go:1856","timestamp":"2024-09-20T13:03:46.888163Z","uid":"c0c6908d-1f15-4f0d-9e9c-c33a1f766439"}

{"component":"virt-launcher","kind":"","level":"info","msg":"**Signaled vmi kill**","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"server.go:327","timestamp":"2024-09-20T13:03:46.888506Z","uid":"c0c6908d-1f15-4f0d-9e9c-c33a1f766439"}
{"component":"virt-launcher","level":"info","msg":"DomainLifecycle event Domain event=\"stopped\" detail=\"destroyed\" with event id 5 reason 1 received","pos":"client.go:470","timestamp":"2024-09-20T13:03:46.888725Z"}
{"component":"virt-launcher","level":"info","msg":"kubevirt domain status: Shutoff(5):Destroyed(2)","pos":"client.go:297","timestamp":"2024-09-20T13:03:46.893138Z"}
{"component":"virt-launcher","level":"info","msg":"Domain name event: kubevirt-test-default1_testvmi-txhzt","pos":"client.go:424","timestamp":"2024-09-20T13:03:46.901339Z"}
{"component":"virt-launcher","level":"info","msg":"DomainLifecycle event Domain event=\"undefined\" detail=\"removed\" with event id 1 reason 0 received","pos":"client.go:470","timestamp":"2024-09-20T13:03:46.910698Z"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Domain undefined.","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"manager.go:1883","timestamp":"2024-09-20T13:03:46.910862Z","uid":"c0c6908d-1f15-4f0d-9e9c-c33a1f766439"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Signaled vmi deletion","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"server.go:363","timestamp":"2024-09-20T13:03:46.911210Z","uid":"c0c6908d-1f15-4f0d-9e9c-c33a1f766439"}
{"component":"virt-launcher","level":"info","msg":"Domain name event: ","pos":"client.go:424","timestamp":"2024-09-20T13:03:46.912914Z"}
{"component":"virt-launcher","level":"info","msg":"Process kubevirt-test-default1_testvmi-txhzt and pid 68 is gone!","pos":"monitor.go:179","timestamp":"2024-09-20T13:03:47.007421Z"}
{"component":"virt-launcher","level":"info","msg":"Waiting on final notifications to be sent to virt-handler.","pos":"virt-launcher.go:281","timestamp":"2024-09-20T13:03:47.008026Z"}
{"component":"virt-launcher","level":"info","msg":"Final Delete notification sent","pos":"virt-launcher.go:296","timestamp":"2024-09-20T13:03:47.008223Z"}
{"component":"virt-launcher","level":"info","msg":"stopping cmd server","pos":"server.go:608","timestamp":"2024-09-20T13:03:47.008343Z"}
{"component":"virt-launcher","level":"info","msg":"cmd server stopped","pos":"server.go:617","timestamp":"2024-09-20T13:03:47.011630Z"}
{"component":"virt-launcher","level":"info","msg":"Exiting...","pos":"virt-launcher.go:512","timestamp":"2024-09-20T13:03:47.011813Z"}
{"component":"virt-launcher-monitor","level":"info","msg":"Reaped pid 7 with status 0","pos":"virt-launcher-monitor.go:198","timestamp":"2024-09-20T13:03:47.026462Z"}



for s390x logs :

"component":"virt-launcher","kind":"","level":"info","msg":"Signaled graceful shutdown for testvmi-txhzt","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"manager.go:1810","timestamp":"2024-09-20T09:23:38.178570Z","uid":"07fd2702-8124-41f2-b393-b8f12a1c61fd"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Signaled vmi shutdown","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"server.go:345","timestamp":"2024-09-20T09:23:38.178829Z","uid":"07fd2702-8124-41f2-b393-b8f12a1c61fd"}
{"component":"virt-launcher","level":"info","msg":"kubevirt domain status: Running(1):Unknown(1)","pos":"client.go:297","timestamp":"2024-09-20T09:23:38.189236Z"}
{"component":"virt-launcher","level":"info","msg":"Received signal terminated","pos":"virt-launcher.go:473","timestamp":"2024-09-20T09:23:38.189222Z"}
{"component":"virt-launcher","kind":"VirtualMachineInstance","level":"info","msg":"Signaled graceful shutdown","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"virt-launcher.go:443","timestamp":"2024-09-20T09:23:38.189319Z","uid":"07fd2702-8124-41f2-b393-b8f12a1c61fd"}
{"component":"virt-launcher","level":"info","msg":"kubevirt domain status: Running(1):Unknown(1)","pos":"client.go:297","timestamp":"2024-09-20T09:23:38.210438Z"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Signaled graceful shutdown for testvmi-txhzt","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"manager.go:1810","timestamp":"2024-09-20T09:23:38.222639Z","uid":"07fd2702-8124-41f2-b393-b8f12a1c61fd"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Signaled vmi shutdown","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"server.go:345","timestamp":"2024-09-20T09:23:38.222683Z","uid":"07fd2702-8124-41f2-b393-b8f12a1c61fd"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Signaled graceful shutdown for testvmi-txhzt","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"manager.go:1810","timestamp":"2024-09-20T09:23:38.236761Z","uid":"07fd2702-8124-41f2-b393-b8f12a1c61fd"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Signaled vmi shutdown","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"server.go:345","timestamp":"2024-09-20T09:23:38.236793Z","uid":"07fd2702-8124-41f2-b393-b8f12a1c61fd"}
{"component":"virt-launcher","level":"info","msg":"DomainLifecycle event Domain event=\"shutdown\" detail=\"unknown\" with event id 6 reason 1 received","pos":"client.go:470","timestamp":"2024-09-20T09:23:42.511281Z"}
{"component":"virt-launcher","level":"info","msg":"kubevirt domain status: ShuttingDown(4):Unknown(0)","pos":"client.go:297","timestamp":"2024-09-20T09:23:42.513074Z"}
{"component":"virt-launcher","level":"info","msg":"Domain name event: kubevirt-test-default1_testvmi-txhzt","pos":"client.go:424","timestamp":"2024-09-20T09:23:42.515171Z"}
{"component":"virt-launcher-monitor","level":"info","msg":"Reaped pid 74 with status 0","pos":"virt-launcher-monitor.go:198","timestamp":"2024-09-20T09:23:42.709392Z"}
{"component":"virt-launcher","level":"info","msg":"DomainLifecycle event Domain event=\"stopped\" detail=\"shutdown\" with event id 5 reason 0 received","pos":"client.go:470","timestamp":"2024-09-20T09:23:42.840383Z"}
{"component":"virt-launcher","level":"info","msg":"kubevirt domain status: Shutoff(5):Shutdown(1)","pos":"client.go:297","timestamp":"2024-09-20T09:23:42.841330Z"}
{"component":"virt-launcher","level":"info","msg":"Domain name event: kubevirt-test-default1_testvmi-txhzt","pos":"client.go:424","timestamp":"2024-09-20T09:23:42.843855Z"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Domain undefined.","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"manager.go:1883","timestamp":"2024-09-20T09:23:42.845477Z","uid":"07fd2702-8124-41f2-b393-b8f12a1c61fd"}
{"component":"virt-launcher","level":"info","msg":"DomainLifecycle event Domain event=\"undefined\" detail=\"removed\" with event id 1 reason 0 received","pos":"client.go:470","timestamp":"2024-09-20T09:23:42.845519Z"}
{"component":"virt-launcher","kind":"","level":"info","msg":"Signaled vmi deletion","name":"testvmi-txhzt","namespace":"kubevirt-test-default1","pos":"server.go:363","timestamp":"2024-09-20T09:23:42.845550Z","uid":"07fd2702-8124-41f2-b393-b8f12a1c61fd"}
{"component":"virt-launcher","level":"info","msg":"Domain name event: ","pos":"client.go:424","timestamp":"2024-09-20T09:23:42.846396Z"}
{"component":"virt-launcher","level":"info","msg":"Process kubevirt-test-default1_testvmi-txhzt and pid 74 is gone!","pos":"monitor.go:179","timestamp":"2024-09-20T09:23:42.851676Z"}
{"component":"virt-launcher","level":"info","msg":"Waiting on final notifications to be sent to virt-handler.","pos":"virt-launcher.go:281","timestamp":"2024-09-20T09:23:42.851728Z"}
{"component":"virt-launcher","level":"info","msg":"Final Delete notification sent","pos":"virt-launcher.go:296","timestamp":"2024-09-20T09:23:42.851755Z"}
{"component":"virt-launcher","level":"info","msg":"stopping cmd server","pos":"server.go:608","timestamp":"2024-09-20T09:23:42.851813Z"}
{"component":"virt-launcher","level":"info","msg":"cmd server stopped","pos":"server.go:617","timestamp":"2024-09-20T09:23:42.852073Z"}
{"component":"virt-launcher","level":"info","msg":"Exiting...","pos":"virt-launcher.go:512","timestamp":"2024-09-20T09:23:42.852098Z"}
{"component":"virt-launcher-monitor","level":"info","msg":"Reaped pid 12 with status 0","pos":"virt-launcher-monitor.go:198","timestamp":"2024-09-20T09:23:42.855862Z"}

Reply:
![Uploading image.png…]()